### PR TITLE
Added Support for downloading JDBC driver from an artifactory which is secured with authentication

### PR DESCRIPTION
**Pull request by Heshma**
Added Support for downloading JDBC driver from an artifactory which is secured with authentication

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,18 @@ RULES_SCHEMA 		| Specify the rules schema for your database. | `rules`
 DATA_SCHEMA 		| Specify the data schema for your database. | `data`
 CUSTOMERDATA_SCHEMA | If configured in your database, set the customer data schema for your database. If you do not provide a value, this setting defaults to `dataSchema`. |
 
+### Custom artifactory settings used for downloading JDBC driver
+
+You can specify custom artifactory authentication details if it is secured with Basic or APIKey Authentication which is used for downloading JDBC driver.
```
Comment (taz-pega-work) [inline README.md]: Configure the Basic or APIKey Authentication details for a custom artifactory authentication you use for managing a JDBC driver for your organization.

**Review by taz-pega-work — APPROVED** [review]
very minor changes to consider

File: scripts/docker-entrypoint.sh [hunk 2]
```diff
@@ -60,17 +65,96 @@ if [ "$JDBC_CLASS" == "" ]; then
...
+fi
+
 if [ "$JDBC_DRIVER_URI" != "" ]; then
+  curl_cmd_options=''
+  if [ "$ENABLE_CUSTOM_ARTIFACTORY_SSL_VERIFICATION" == true ]; then
+    echo "Establishing a secure connection to download driver."
+    curl_cmd_options="curl -sSL $custom_artifactory_auth $custom_artifactory_certificate"
+  else
+    echo "Establishing an insecure connection to download driver."
+    curl_cmd_options="curl -ksSL $custom_artifactory_auth"
```
Comment (petejo) [inline scripts/docker-entrypoint.sh:145]: Now that we can import certs prior to the curl call, we should drop support for the -k flag.

**Review by taz-pega-work — APPROVED** [review]

File: scripts/docker-entrypoint.sh (hunk 2 above)
Comment (Heshma) [reply scripts/docker-entrypoint.sh:145]: @petejo From the pega-helm-charts ENABLE_CUSTOM_ARTIFACTORY_SSL_VERIFICATION is always defaulted to true, in which case we try to make only secure connections.
In cases where client doesn't want to provide a certificate or certificate is expired, one can use this option to skip SSL verification.  Considering above cases, we have added this option where client can disable SSL verification, do you think we have to drop the support?

**Review by pega-talba — APPROVED** [review]

**Review by misterdorito — APPROVED** [review]

**Pull request merged**